e2e-tests/desktop/tests/credentials.spec.js

```diff
@@ -0,0 +1,310 @@
+      {
+        encoding: 'utf-8',
+      },
+    );
+
+    execSync(`vault policy write ${secretPolicyName} ${kvPolicyPath}`);
+    const vaultToken = JSON.parse(
+      execSync(
+        `vault token create` +
+          ` -no-default-policy=true` +
+          ` -policy=${boundaryPolicyName}` +
+          ` -policy=${secretPolicyName}` +
+          ` -orphan=true` +
+          ` -period=20m` +
+          ` -renewable=true` +
+          ` -format=json`,
+      ),
+    );
+    const clientToken = vaultToken.auth.client_token;
+
+    // Create a vault credential store and generic secrets type credential library
+    const vaultCredentialStore = await boundaryHttp.createVaultCredentialStore(
+      request,
+      project.id,
+      vaultAddr,
+      clientToken,
+    );
+    credentialLibrary = await boundaryHttp.createVaultCredentialGenericSecret(
+      request,
+      {
+        credentialStoreId: vaultCredentialStore.id,
+        vaultPath: `${secretsPath}/data/${secretName}`,
+      },
+    );
+
+    // Create a tcp target with vault brokered credential Libraries
+    targetWithBrokeredVaultCredentials = await boundaryHttp.createTarget(
+      request,
+      {
+        scopeId: project.id,
+        type: 'tcp',
+        port: targetPort,
+      },
+    );
+    targetWithBrokeredVaultCredentials = await boundaryHttp.addHostSource(
+      request,
+      {
+        target: targetWithBrokeredVaultCredentials,
+        hostSourceIds: [hostSet.id],
+      },
+    );
+    targetWithBrokeredVaultCredentials =
+      await boundaryHttp.addBrokeredCredentials(request, {
+        target: targetWithBrokeredVaultCredentials,
+        credentialIds: [credentialLibrary.id],
+      });
+
+    // Create a tcp target with static brokered credentials
+    targetWithBrokeredStaticCredentials = await boundaryHttp.createTarget(
+      request,
+      {
+        scopeId: project.id,
+        type: 'tcp',
+        port: targetPort,
+      },
+    );
+    targetWithBrokeredStaticCredentials = await boundaryHttp.addHostSource(
+      request,
+      {
+        target: targetWithBrokeredStaticCredentials,
+        hostSourceIds: [hostSet.id],
+      },
+    );
+    targetWithBrokeredStaticCredentials =
+      await boundaryHttp.addBrokeredCredentials(request, {
+        target: targetWithBrokeredStaticCredentials,
+        credentialIds: [sshPrivateKeyCredential.id, jsonCredential.id],
+      });
+
+    targetWithInjectedCredentials = await boundaryHttp.createTarget(request, {
+      scopeId: project.id,
+      type: 'ssh',
+      port: targetPort,
+    });
+    targetWithInjectedCredentials = await boundaryHttp.addHostSource(request, {
+      target: targetWithInjectedCredentials,
+      hostSourceIds: [hostSet.id],
+    });
+    targetWithInjectedCredentials = await boundaryHttp.addInjectedCredentials(
+      request,
+      {
+        target: targetWithInjectedCredentials,
+        credentialIds: [sshPrivateKeyCredential.id],
+      },
+    );
```

Comment:
Maybe I missed this but what is `targetWithInjectedCredentials` used for?

Reply:
I originally moved this from targets, but I moved it back. But I will have to check if injected credentials are also displayed in the credential panel for ssh targets

Reply:
this is not needed, we only display brokered credentials in the cred panel in desktop, I have removed these from credentials test file.